### ISSUE TITLE
DirecTV

### ISSUE THREAD

**Issue by takeachillphill** [n00neimp0rtant/xCon-Issues#507]
Both the Directv and GenieGo app are not supported. 
![image](https://cloud.githubusercontent.com/assets/13226935/8557808/baf2bd8e-24cd-11e5-9c97-35ce6881bf60.jpg)
![image](https://cloud.githubusercontent.com/assets/13226935/8557809/baf81e00-24cd-11e5-8437-8fdc573adac5.jpg)
![image](https://cloud.githubusercontent.com/assets/13226935/8557810/baf833f4-24cd-11e5-8066-2d82baccb2b5.jpg)

**Comment by KOTULCN** [comment]
What devices and version of xCon are you using? Beta 5 is working for me on iPhone 6, iPhone 5s, and iPad 4.

**Comment by takeachillphill** [comment]
iPhone 6 on xCon version 38? Whichever is the latest version of xCon as I just installed it

**Comment by KOTULCN** [comment]
Install the beta 5 that's located in the readme link on main xCon page.

**Comment by takeachillphill** [comment]
Beta 5 worked! Thank you so much for your help

**Comment by trj111** [comment]
Beta 5 not working for me. iPhone 5. iOS 8.3

**Comment by trj111** [comment]
Elaborate on issue. 

It bypasses jailbreak detection message but it just says Initializing with a spinning wheel FOREVER.

**Comment by icebox766** [comment]
I have the curent version of directv on an iPhone 5 running 8.4 jb, and directv works for me with Xcon 40b5.

**Comment by trj111** [comment]
How long did it take to initialize?

**Comment by icebox766** [comment]
All initializing and loading takes 10 seconds before the app becomes usable.

**Comment by trj111** [comment]
Then it isn't working on 8.3 jb

**Comment by JEmlay** [comment]
This has been broken on iPhone6 for a while now.  I just upgraded from iOS 8.1 to iOS 8.4 and loaded xCon v40 b5.  I get the same thing as before, it says Initializing and does that forever.  So it's definantly not the iOS version or the jailbreak in question.  Remove xCon and I get the jailbreak message.

As always, works fine on my iPad Air.

**Comment by KOTULCN** [comment]
I don't know what to tell you I've been using it just fine on 8.4. You tried deleting the app and reinstalling it? If that doesn't work maybe you have another tweak that is doing something crazy. Why not backup and use the new impactor tweak and re-jailbreak installing only xcon and see if that works? There are other ways to disable individual daemons but that is more advanced and I dont want to screw you up.

**Comment by JEmlay** [comment]
Every time I've jailbroken a device I install xCon first and test it by itself.  Happens every time.  I'm also not the only one.  There's another thread somewhere with other iPhone6 users having the same exact problem.  I just came here to report the same thing on 8.4.

**Comment by KOTULCN** [comment]
Thats odd. I know you could use Flex 2 on iOS 8.1.2. Maybe you could try that until xCon is updated?

**Comment by trj111** [comment]
I am trying Flex 2 as well but it doesn't do anything different.

It bypasses the jailbreak detection but the DirecTV app hangs on
"Initializing".  Perhaps I'm not the most patient person in the world, but
after 10 minutes I gave up waiting for it to initialize.

**Comment by JEmlay** [comment]
There's a whole thread on Flex 2 as well.  For some it doesn't show any available apps making it useless.  For others it shows the DTV app but doesn't do anything different like trj111 just mentioned.

**Comment by KOTULCN** [comment]
Try installing a previous version of the directv app?

**Comment by JEmlay** [comment]
I guess I could try that but in the past DTV has blocked older version.

End result is...this hasn't worked on our phones since our iPhone 5's.  So it's been a while.  There has never been a fix for it.  We just keep on waiting.  See what happens with the iPhone 7's.

**Comment by takeachillphill** [comment]
xCon has worked for me since I followed the advice from another user.

**Comment by trj111** [comment]
What advice?

**Comment by takeachillphill** [comment]
KOTULCN: "Install the beta 5 that's located in the readme link on main xCon page."

**Comment by trj111** [comment]
That didn't work.

**Comment by takeachillphill** [comment]
I spoke too soon. The DirecTV app is no longer working for me

**Comment by KOTULCN** [comment]
The latest directv update ( 4.2.023) has broken support with xCon. The app just sits on "initializing."

**Comment by JEmlay** [comment]
Thanks for the heads up.  That's how it's been on the iPhone for many of us for a long while now.  I backed up my current iPad version before updating and I can confirm that the iPad version is now exactly like my iPhone version.....totally hosed and hangs in initializing.  Luckily they are still allowing us to login with the old version.  But that probably won't last long.

**Comment by KOTULCN** [comment]
I wonder if we'll ever see an update?

**Comment by trj111** [comment]
Can we download the older version of directv that still works with xcon?

**Comment by KOTULCN** [comment]
I backed up my iPhone 6 then did a DFU restore, before restoring from backup I installed the Directv app, messed around with it, then restored from backup in iTunes. Once restoration of apps/settings was completed I jailbroke. After setting Cydia back up for the first time I installed iFile and installed xCon beta 5 (latest update) and the Directv app works just fine. This is the newest version of the app.

**Comment by drocco21** [comment]
I tried the beta5 but Genie Go is being a jerk, same error as shown above :/. iPhone 6 with 8.4

**Comment by JEmlay** [comment]
The old version of DTV was cut off (iPad).  So I upgraded just to get rid of the update notice.  Same issue.  So I deleted it then reinstalled it.  After that is was fine.  WTF?  That's the second time I've had to do that.

At any rate the iPhone version is still borked showing initializing.  Go in safe mode and it runs fine.

**Comment by KOTULCN** [comment]
Had an issue where app quite seeing my account so I deleted and reinstalled......stuck on initializing again.

**Comment by JEmlay** [comment]
DTV just updated the iPad app, AGAIN.  It still works (v4.2.05).  iPhone version as always doesn't work (4.2.023).

**Comment by KOTULCN** [comment]
Just updated to the latest DTV revision and app works using tsProtector 8. I am not sure if xCon beta 5 works but if someone wants to try it out and post I am sure it would be appreciated.

**Comment by JEmlay** [comment]
I take it you mean DTV for iPhone v4.2.103

Nope, doesn't work.  Stuck at initializing.  iPhone 6 iOS 8.4 using xCon v40b5

**Comment by KOTULCN** [comment]
Yes I was talking about that update. did you re-install xCon and respring after updating? If that doesn't work try tsProtector 8.

**Comment by JEmlay** [comment]
Yes, I reinstalled everything.  I do that now since I've had to do it twice on my iPad.

**Comment by trj111** [comment]
Tsprotector8 asks me to update to the latest version of directv app

**Comment by JEmlay** [comment]
I finally found a version of tsProtector that would install on my devices.  It works with DTV on my iPad just like xCon did.  However on my iPhone6 I get the same exact thing, the app is stuck at INITIALIZING.  So what does this have to do with xCon?  Well, whatever that INITIALIZING issue is, it's not only with xCon.  tsProtector is doing the same exact thing.  Does it on both my iPhone6 and iPhone6 Plus.

**Comment by Docnim** [comment]
iPad and iPhone apps both updated again. Anyone try the updates yet? I am staying on previous versions on both devices, both still working fine with xcon 40 b5. Thanks!

**Comment by Docnim** [comment]
I updated to latest Directv apps on both iPad and iPhone 6plus 8.4 
Both work under xcon 40 b5

**Comment by Docnim** [comment]
New Directv App 4.2.121 for the iPad today.  Anyone try it out yet?

**Comment by Docnim** [comment]
I couldn't wait.  I updated to 4.2.121 on the iPad. Xcon 40 b5 still working well !

**Comment by Maestro64** [comment]
when people say it is still working, make sure you state the IOS version as well

IOS 9.0.2
Xcon 40 Beta 6
DirecTV 4.2.121 Not working

I know it was working on IOS 8.4 since i inadvertently update to 4.2.121 before I update to 9.0.2. after the update DTV says it does not work on a jailbroken device.

Be warned ,

**Comment by JEmlay** [comment]
I was never able to get DTV to work on my iPhones since iOS8 (iPad yes).  For some odd reason I couldn't get Flex2 to work either. Now with iOS 9.0.2 Flex2 works great.  I have DTV working on all my iDevices with Flex2.

**Comment by loafin** [comment]
For anyone who uses a hosts blocker, make sure it's not blocking "googleads.g.doubleclick.net".  I had the "Initializing..." issue on iOS 8/9 with the more recent iPhone versions of the app when used with Minimal Hosts Blocker.  Commenting out this specific host entry allows me to log in, even in the latest version 4.2.202.  (I'm using tsProtector nowadays but I think this would still apply to xCon, or even Flex.)

**Comment by azroach** [comment]
I have iOS 9.0.2 and xcon beta 7 and directv app still says my device is jailbroken.

**Comment by johnsoninnyc** [comment]
wrong beta --  follow instructions I posted a few days back under the Good For Enterprise thread...

**Comment by azroach** [comment]
@johnsoninnyc @loafin

Thanks it worked.  But it looks like I did have the right beta because it works with xcon beta 7.

After entering the command:

ldid -S /Library/MobileSubstrate/DynamicLibraries/xCon.dylib

And respringing my device, xcon beta 7 works to bypass jailbreak detection for the directv app

Also had to make sure my hosts blocker wasn't blocking the following:

googleads.g.doubleclick.net

**Comment by johnsoninnyc** [comment]
good to hear!  and you will be fine for 64-bit apps ; but for any 32-bit apps beta 7 won't work  because it wasn't compiled properly for iOS 9.  beta eight fixes that.    also for any older 32-bit devices I wouldn't think the beta seven would work on anything FYI.

**Comment by azroach** [comment]
ok, cool.  so what repo can i get beta 8 from then?

**Comment by johnsoninnyc** [comment]
There's a link under tHe good for enterprise thread and also under the "iOS 9.0.2 iPhone 5s safe mode" thread     Both are threads here on github under xcon project/n00neimp0rtant   Read the threads please.

**Comment by JEmlay** [comment]
Beta 7?  Beta 8?  Why does this project only go up to Beta 6?  Why are we having to read through random threads that have nothing to do with us just to get information?  I don't have a 5S and I don't use GFE so it makes no sense for me to ever even look at those threads.  This project is wasting away.

**Comment by trj111** [comment]
Beta 8 worked for me!

I removed my host blocker and voila. 

Now can I reload the host blocker?

**Comment by JEmlay** [comment]
You can reload it but make sure that URL isn't blocked.

**Comment by azroach** [comment]
@JEmlay

it goes up to beta 8 obviously.  lol i do understand what you mean by having to sift through all the threads to get what you need.  that's why i replied with the exact information that i found that helped me solve my issue.

it would help if i was provided the url of the repo that i need to find xcon beta 8, but i guess ill go look for it anyway.

**Comment by azroach** [comment]
@johnsoninnyc 

i found the link for beta 8 (https://goo.gl/rcGmgv) but is there a beta repo?

i tried adding the beta repo in the good for enterprise thread but it doesn't seem to be valid

**Comment by JEmlay** [comment]
@azroach

"it goes up to beta 8 obviously"

Like I said....https://github.com/n00neimp0rtant/xCon-Issues

Thus...having to find these rogue betas in various threads from external links.

**Comment by azroach** [comment]
yea it's pretty unorganized in my opinion

**Comment by azroach** [comment]
looks like there's a lack of leadership or something...

it would be nice if their beta repo was actually working and it was displayed somewhere you can easily see it.

and the instruction on how to code sign the damn tweak should be posted on that first "xcon-issues" page because if you don't do that then the stupid tweak doesn't even work.

pretty lazy and sloppy work to tell you the truth.

**Comment by JEmlay** [comment]
That's why I moved onto Flex.  Not only does it get the job done but you can use it for so much more AND I no longer have all the crashing issues that I used to have once I removed xCon.

**Comment by azroach** [comment]
if you're trying to bypass jailbreak detection, then xcon is a lot better than flex because i always had to either reboot my device or run icleaner pro every time i wanted to use the directv app.

with xcon, it always bypasses the jailbreak detection and you don't need to always reboot your device or run icleaner pro to enable it.

---

did you try beta 8 to see if you get any crashing issues?

**Comment by JEmlay** [comment]
Flex works 100% of the time.  I never have to reboot anything.  Plus all the crashing issues that have disappeared.  I also use Flex and have never even heard of icleaner.

It just depends on what all you're running on your device.

**Comment by azroach** [comment]
glad flex works for you.  i'm using a flex patch for rhapsody and i always have to run icleaner pro and then respring my device to enable the patch ever so often.

and the same thing was happening for directv.

happened to me on two devices so far.  hard to say what is causing the issue, but it seems to be fixed if i just use xcon for directv and a .deb file for rhapsody

**Comment by johnsoninnyc** [comment]
agree re: user unfriendly approach , but for those of us too impatient for more official versions , this workaround is sufficient. 

re: repo -- forget repos -- just install ifile from cydia. then open the xcon deb file in ifile and then choose "Installer" and it will install fine (provided that Cydia substrate is already installed). --   then run the code sign trick which you only have to do once and you should be all set after respring. ..

**Comment by azroach** [comment]
looks like it's stuck now on "initializing"

and i did make sure my hosts blocker wasn't blocking "googleads.g.doubleclick.net"

any ideas what's going on?

**Comment by JEmlay** [comment]
Open your browser and try to hit googleads.g.doubleclick.net

See if it answers.  It should redirect you.

**Comment by azroach** [comment]
ok let me check

**Comment by azroach** [comment]
says safari can't open the page.

**Comment by azroach** [comment]
should i unblock "googleads2.g.doubleclick.net" as well?

**Comment by azroach** [comment]
i removed the hosts blocker and now it works.

anybody know what exactly i should not block? i don't want to remove all the hosts that i am blocking...

**Comment by filthyjason** [comment]
You can also use Mobile terminal to download / install if you've added Debian Packager. Just:
wget https://goo.gl/rcGmgv --no-check-certificate
dpkg -i rcGmgv
ldid -S /Library/MobileSubstrate/DynamicLibraries/xCon.dylib
respring
I had renamed the rcGmgv to xcon40b8.deb when I did this, not sure if it's required to be a .deb file. If you have any problems, then try it like this:
wget https://goo.gl/rcGmgv --no-check-certificate
mv rcGmgv xcon40b8.deb
dpkg -i xcon40b8.deb
ldid -S /Library/MobileSubstrate/DynamicLibraries/xCon.dylib
respring

Any idea what's involved in hosting a repo? I like & use xCon & I could see if this is something I'd be able to host for the community.

**Comment by azroach** [comment]
@filthyjason 

what iOS are you using?  mobile terminal doesn't workin iOS 9.0.2...

**Comment by filthyjason** [comment]
@azroach I have been using a copy from Karen's Pineapple Repo:
http://cydia.angelxwind.net
for a year at least. She has version 521-karen-1 and it runs fine on my iPhone 6 on 9.0.2.

**Comment by azroach** [comment]
what's it called?  i have that repo too and i've never seen a mobile terminal

**Comment by filthyjason** [comment]
MobileTerminal
There's no space in the name. If you go to Sources in Cydia, then to her repo, then Terminal Support, it'll be the only one in there.

**Comment by azroach** [comment]
ok cool thanks.  looks like you can get xcon version 41 now from jailbreakvideo.ru repo and it works without having to code sign it yourself.

i just uninstalled my hosts blocker, logged into directv, then reinstalled the hosts blocker and directv works fine.

**Comment by azroach** [comment]
i think this issue can be closed now...

**Comment by Jrisc96** [comment]
After 30 mins of trying to figure out how to fix this downloaded xcon VERSION 41 (the only one that works) and it worked perfectly, thank you so much azroach!

**Comment by naukarialerts4u** [comment]
directv app is a very good app for watching our favorite movies and tv shows. and at&t subscription was good. and most good thing is it is available for all platforms thank you for this useful and helpful guide about directv app 
https://goo.gl/7dR8YN
